ui/components/app/confirm/title/useDescription.ts

```diff
@@ -0,0 +1,24 @@
+import { useSelector } from 'react-redux';
+import { TransactionType } from '@metamask/transaction-controller';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../selectors';
+
+// Map transaction type to translation keys
+const translationMap: Partial<Record<TransactionType, string>> = {
+  [TransactionType.personalSign]: 'confirmTitleDescSignature',
+};
+
+/** Hook to get the description based on the transaction type */
+const useDescription = (): string => {
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+
+  if (!currentConfirmation) {
+    return '';
+  }
+
+  const translationKey = translationMap[currentConfirmation.type];
+  return t(translationKey);
+};
+
+export default useDescription;
```

Comment:
Hey @digiwand , we may not need to create hook is logic is small and can be included in component itself. The 2 hooks have code repetition and code can go into the component itself its small.

Reply:
Hey @jpuri,

my thought here was keeping the hook logic separate from the presentational component could be a good pattern to use.

It also looks like the description may have more variants in the future e.g. petname support 
https://www.figma.com/file/CeAIcptzeaI5ciTpYBaiqr/Confirmations-DS?type=design&node-id=853-41548&mode=design&t=fVyvrkgypS75B9Jt-0

I considered using 1 hook for both of them and went with keeping them separate since I don't imagine the performance difference would be significant

ref: [React pattern blog](https://www.patterns.dev/react/hooks-pattern/#:~:text=React%20Hooks%20are%20functions%20that,such%20as%20componentDidMount%20and%20componentWillUnmount)

Reply:
updated based on discussion. The file is no longer using separate hooks. The logic lives in these files for @jpuri's reason about and to have less files to maintain https://github.com/MetaMask/metamask-extension/pull/22098/commits/73f98cc0e3720f19f0d4d6e0dfcb340f65812c83